README.md

```diff
@@ -2,9 +2,64 @@
 # help-to-save-kalculator
 
 ##### code coverages
-![LINE](https://img.shields.io/badge/line--coverage-98%25-brightgreen.svg)
-![BRANCH](https://img.shields.io/badge/branch--coverage-85%25-brightgreen.svg)
-![COMPLEXITY](https://img.shields.io/badge/complexity-1.40-brightgreen.svg)
+![LINE](https://img.shields.io/badge/line--coverage-92%25-brightgreen.svg)
+![BRANCH](https://img.shields.io/badge/branch--coverage-89%25-brightgreen.svg)
+![COMPLEXITY](https://img.shields.io/badge/complexity-1.59-brightgreen.svg)
+
+## Calculate help to save bonus
+
+```kotlin
+Calculator.run(
+                regularPayment = 50,                // Required (must be between 1 and 50)   
+                currentBalance = null,              // Optional (Int) (Default: null)
+                currentFirstPeriodBonus = null,     // Optional (Double) (Default: null)
+                currentSecondPeriodBonus = null,    // Optional (Double) (Default: null)
+                accountStartDate = null             // Optional (DateTime) (Default: null)
+)          
+```
+
+ > `currentBalance`, `currentFirstPeriodBonus`, `currentSecondPeriodBonus`, `currentSecondPeriodBonus` and `accountStartDate` must either be ALL provided or all null.
```

Comment:
if this is the case, could enforce this by having two public `run` methods with signatures:
```
fun run(regularPayment: Int)
```
and 
```
fun run(
         regularPayment: Int,
         currentBalance: Int,
         currentFirstPeriodBonus: Double,
         currentSecondPeriodBonus: Double,
         accountStartDate: DateTime)
```
which internally call a private:
```
fun run(
         regularPayment: Int,
         currentBalance: Int? = null,
         currentFirstPeriodBonus: Double? = null,
         currentSecondPeriodBonus: Double? = null,
         accountStartDate: DateTime? = null)
```

Reply:
also this would remove the issue with default values on iOS as noted in the README below